### ISSUE TITLE
Creating an HDInsight object with a Datalake Gen1 as primary storage (feature parity with REST/WebUI)

### ISSUE THREAD

**Issue by valenca** [Azure/azure-cli#14835]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
`az hdinsight create` returns error `If storage details are specified, the storage account, storage account key, and the default container must be specified` when no explicit storage flags are given. The attempt here is to use a Datalake Gen1 as the primary storage. This is doable via UI (and I suspect REST API), but not via the CLI tool.

**To Reproduce**
Running the following command (with bash variables replaced with sensible values):
```
az hdinsight create \
    -n ${HDINSIGHT} \
    -g ${RESOURCE_GROUP} \
    --subnet ${SUBNET} \
    --vnet-name ${VNET} \
    --headnode-size Standard_A5 \
    --workernode-size Standard_A5 \
    --workernode-count 1 \
    --version 3.6 \
    --type hadoop \
    --http-password ${PASSWORD} \
    --cluster-configurations '{"gateway":{"restAuthCredential.isEnabled":true,"restAuthCredential.username":"admin","restAuthCredential.password":"${PASSWORD}"},"core-site":{"fs.defaultFS":"adl://home","dfs.adls.home.hostname":"${DATALAKE}.azuredatalakestore.net","dfs.adls.home.mountpoint":"/"},"clusterIdentity":{"clusterIdentity.applicationId":"${APP_ID}","clusterIdentity.certificate":"${CERTIFICATE}","clusterIdentity.aadTenantId":"https://login.windows.net/${TENANT_ID}","clusterIdentity.resourceUri":"https://datalake.azure.net/","clusterIdentity.certificatePassword":"12345"}}' \
    --tag ${PREFIX} \
    --debug | tee resources/${HDINSIGHT}.json
```
Omitting the "cluster-configurations" throws the same error. Giving the Microsoft.DatalakeStore id through the --storage-account flag is invalid as only Microsoft.Storage objects seem to be accepted.

**Expected behavior**
Either a more appropriate error message or a DatalakeStore object/ID to be an acceptable input for the command.

**Environment summary**

**Comment by yonzhan** [comment]
hdinsight

**Comment by aim-for-better** [comment]
Thanks for feedback, We will investigate it soon.

**Comment by aim-for-better** [comment]
> For what it's worth for anyone else looking: I found a workaround for this by using az resource create -p and feeding it the properties in a json file. It's not as neat as the rest of the script that uses az cli commands, but it did the job. To get the properties field, I configured an HDInsight directly in the UI, intercepted the "create" request, and replaced all the values there with the values I needed (be careful not to look for it in the validate request like I did for a while).

Thanks for sharing.

**Comment by aim-for-better** [comment]
This is an issue for CLI. We will fix it in Oct or Nov.

**Comment by deshriva** [comment]
@valenca : This is still in our backlog and we will continue to provide monthly updates until the issue is addressed.

**Comment by aim-for-better** [comment]
@valenca , this issue is fixed in the latest azure cli version 2.14.0. Thanks for feedback.
